app/controllers/comments_controller.rb

```diff
@@ -16,6 +17,18 @@ def create
     redirect_to user_post_path(current_user, @comment.post_id)
   end
 
+  def destroy
+    @comment = Comment.find(params[:id])
+    authorize! :destroy, @comment
+    redirect_url = request.referer || fallback_url
+
+    if @comment.destroy
+      redirect_to redirect_url, notice: 'Comment was successfully deleted.'
```

Comment:
**OPTIONAL** - There is an issue with your code when creating a comment, in the strong parameters you have used, you are permitting `text` but in your schema, you have labeled the text attribute as `body` instead. Please correct the attribute used in your strong parameters. Please fix this 🙏